base/linalg/symmetric.jl

```diff
@@ -266,9 +266,9 @@ for (funm, func) in ([:logm,:log], [:sqrtm,:sqrt])
         function ($funm)(A::Symmetric)
             F = eigfact(A)
             if isposdef(F)
-                retmat = (F.vectors * Diagonal(($func)(F.values))) * F.vectors'
+                retmat = (F.vectors * Diagonal(($func).(F.values))) * F.vectors'
             else
-                retmat = (F.vectors * Diagonal(($func)(complex(F.values)))) * F.vectors'
+                retmat = (F.vectors * Diagonal(($func).(complex(F.values)))) * F.vectors'
```

Comment:
Can't we now do `($func).(complex.(F.values))`, and this will be better anyway because it will fuse the loop?

Should we get rid of `float` and `complex` for arrays in favor of dot calls?